src/Sarif/ExtensionMethods.cs

```diff
@@ -14,9 +14,15 @@ namespace Microsoft.CodeAnalysis.Sarif
 {
     public static class ExtensionMethods
     {
-        public static bool HasAtLeastOneNonNullValue(this IList<string> list)
+        public static bool HasAtLeastOneNonNullValue<T>(this IEnumerable<T> collection)
         {
-            return list != null && list.Any((s) => s != null);
+            return collection != null && collection.Any((m) => m != null);
+        }
+
+        public static bool HasAtLeastOneNonDefaultValue<T>(this IEnumerable<T> collection, IEqualityComparer<T> comparer) where T : new()
```

Comment:
>HasAtLeastOneNonDefaultValue [](start = 27, length = 28)

Note that we could eliminate the argument from this helper if ValueEquals was a member of ISarifNode. This would require a generic interface, however.